wordpress-mobile/WordPress-iOS
# Prompts: Add Bloganuary dashboard card

Closes #22120 

This adds a dashboard card entry for Bloganuary. We're only showing the card within December 2023 (for simplicity's sake, since we are targeting the beta) and only showing it to sites that are marked as blogging sites from the backend (i.e. `BloggingPromptSettings.isPotentialBloggingSite`). For more details about the card visibility logic, refer to pctCYC-16J-p2#comment-1131.

Note that I've also added a new remote feature flag in this PR. The card should not be visible when the flag is off.

> [!WARNING]
> The Learn more button is not yet implemented along with the Tracks for the card. The button functionality will be added in the next PR, along with the modal implementation. Tracks will be added at the end.

Preview: 

Light | Dark
-|-
![card-light](https://github.com/wordpress-mobile/WordPress-iOS/assets/1299411/8ad4b812-454d-471f-a8f3-4a2b2ee76f3a) | ![card-dark](https://github.com/wordpress-mobile/WordPress-iOS/assets/1299411/c14db0ab-de7d-4f03-a059-c022b057e8ae)

## To test

> [!NOTE]
> Since the logic involves checking for the current date, I'd suggest modifying the code a little to ensure that the date checking part is skipped. Not to worry though, I've added unit tests for the date checking part! 🙂 
>
> You can temporarily comment out `isDateInDecember` closure and always return `true`:
> ```swift
> let isDateInDecember = true
> ```
> 
> Alternatively, you can also change your device date to December 2023.

- Launch the Jetpack app. 
- Log in with your WP.com account.
- Turn on the `bloganuaryDashboardNudge` feature flag.
- Relaunch the app.
- 🔎  Verify that the Bloganuary card is shown.
- Tap on the ellipsis menu and hide the card.
-  🔎  Verify that the card is now hidden.

Some other variations are covered in the unit test. Please run them and verify that they pass.

## Regression Notes
1. Potential unintended areas of impact
Should be none. The card is a new component and is gated behind a remote feature flag.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manually tested the changes.

3. What automated tests I added (or what prevented me from doing so)
Added unit tests to cover several scenarios that could affect card visibility.

PR submission checklist:


UI Changes testing checklist:
